app/Http/Controllers/Manager/CampaignsController.php

```diff
@@ -237,8 +253,15 @@ public function edit(Request $request, int $campaignId): JsonResponse
         );
 
         $input = $request->input('campaign');
-        $input['targeting_requires'] = $request->input('campaign.targeting.requires');
-        $input['targeting_excludes'] = $request->input('campaign.targeting.excludes');
+
+        $targetingProcessor = new TargetingProcessor($this->configurationRepository->fetchTargetingOptions());
+
+        $input['targeting_requires'] = $targetingProcessor->processTargeting(
+            $request->input('campaign.targeting.requires')
+        );
+        $input['targeting_excludes'] = $targetingProcessor->processTargeting(
+            $request->input('campaign.targeting.excludes')
+        );
```

Comment:
The same code block is used twice. I'd suggest to move it to a dedicated method (``` $input = $this->processTargeting($input) ```).

Reply:
sure